pkg/actors/actors.go

```diff
@@ -937,6 +955,18 @@ func (a *actorsRuntime) CreateReminder(ctx context.Context, req *CreateReminderR
 		}
 	}
 
+	// Store the reminder in active reminders list
+	actorKey := a.constructCompositeKey(req.ActorType, req.ActorID)
+	reminderKey := a.constructCompositeKey(actorKey, req.Name)
+	stop := make(chan bool, 1)
+	id, _ := rand.Int(rand.Reader, big.NewInt(5000))
+	reminderID := int(id.Int64())
+	activeReminder := ActiveReminder{
+		ID:       reminderID,
+		StopChan: stop,
+	}
+	a.activeReminders.Store(reminderKey, activeReminder)
```

Comment:
1. If I understand correctly, with this approach, user app or sdk needs to maintain reminder id returned from dapr runtime. Am I right? 

2. when it comes to id generation, I would use uuid instead of using rand.Int. 

3. Let's go back to the root cause of this bug. 

This bug happens when user app creates the same reminders in a row without delay. the goroutine for each reminder inside createreminder was not stopped correctly by DeleteReminder call because stopchan channel value from map could be new one by one of the requested same reminders without ordering.

So can we just use mutex to reminder object to lock the current reminder before DeleteReminder in createReminder instead of generation id? then it maintains the order of reminder creation requests. stopchan will be always valid. This change will be simpler than this PR without breaking change.

Reply:
In short, we can resolve this issue by adding one lock around deletion without any additional code. single lock will not impact performance as well because deletion and addition of reminder and timer doesn't take too long. 

https://github.com/dapr/dapr/blob/19e6d3256f62a46c0a3df1b612b7081336458e9a/pkg/actors/actors.go#L924-L938

Reply:
I agree, looks like a Lock there should fix it.

Reply:
@youngbupark - As discussed, I don't think a lock will fix the issue. Let me see if I can think of a simpler way to fix it. To clarify, the SDK code is not modified at all. This is a breaking change only because the users might have persistent reminders which will not have an ID and when they use this new code, it will expect an ID. So old reminders will not work.